hack/benchmark/assets/x1b/loader_bench_test.go

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package x1b
+
+import (
+	"testing"
+)
+
+const (
+	bvecsFile = "../large/sift1b/bigann_base.bvecs"
+	fvecsFile = "../large/sift1b/gnd/dis_1000M.fvecs"
+	ivecsFile = "../large/sift1b/gnd/idx_1000M.ivecs"
+)
+
+func BenchmarkBVecs(b *testing.B) {
+	bv, err := NewUint8Vectors(bvecsFile)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer func() {
+		if err := bv.Close(); err != nil {
+			b.Fatal(err)
+		}
+	}()
+
+	i := 0
+	b.Run(bvecsFile, func(bb *testing.B) {
+		bb.ReportAllocs()
+		bb.ResetTimer()
+		for n := 0; n < bb.N; n++ {
+			_, err := bv.Load(i)
+			switch err {
+			case nil:
+				i++
+			case ErrOutOfBounds:
+				i = 0
+			default:
+				bb.Fatal(err)
+			}
+		}
+	})
```

Comment:
I think it has the same problem: some data may lost when error occurs.

```suggestion
	b.Run(bvecsFile, func(bb *testing.B) {
		bb.ReportAllocs()
		bb.ResetTimer()
		
		i := 0
		for n := 0; n < bb.N; {
			_, err := bv.Load(i)
			switch err {
			case nil:
				i++
				n++
			case ErrOutOfBounds:
				i = 0
			default:
				bb.Fatal(err)
			}
		}
	})
```

Reply:
If the error "ErrOutOfBounds" is occurred, the whole data is loaded and it should continue the benchmark from the head.
If the others are occurred, they stops the benchmark.

Reply:
Sorry if i am wrong here.
The problem is that the `n++` will still be executed even when the error occurred, so the number of times `bv.Load` executed maybe less than `bb.N`.

Reply:
I see. I got a good idea for this code.
In this code ErrOutOfBounds is never occurred and I think it seems easy to understand the codes.
However, % operation takes much time.
```
		i := 0
		size := bv.Size()
		for n := 0; n < bb.N; n++ {
			_, err := bv.Load(i % size)
			if err != nil {
				bb.Fatal(err)
			}
			i++
		}
```

Reply:
I think it is good!